### PR TITLE
Remove Gravatar related deprecated types and replace with Gravatar SDK

### PR DESCRIPTION
**Pull request by pinarol**
Fixes #

We deprecated some Gravatar related types and methods when integrating the Gravatar SDK with PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/22921. Now it is time to remove those deprecated types.

This PR also replaces a piece of code in WordPressAuthenticator that was still using the deprecated methods.

To test:

The avatar should look OK in the password entry page:

## Regression Notes
1. Potential unintended areas of impact

Avatars in login flow.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I tested the login flow.

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [x] WordPress.com sites and self-hosted Jetpack sites.
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [x] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages in Xcode.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/296">25.4</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPressAuthenticator/Sources/Signin/LoginLinkRequestViewController.swift [hunk 1]
```diff
@@ -36,7 +37,9 @@ class LoginLinkRequestViewController: LoginViewController {
 
         let email = loginFields.username
         if email.isValidEmail() {
-            gravatarView?.downloadGravatarWithEmail(email, rating: .x)
+            Task {
+                try await gravatarView?.setGravatarImage(with: email, placeholder: .gravatarPlaceholderImage, rating: .x)
```
Comment (pinarol) [inline WordPressAuthenticator/Sources/Signin/LoginLinkRequestViewController.swift]: I wasn't able to make this code run even though I followed the email link flow. Maybe I am missing something.

But this is the same method we call from `GravatarEmailTableViewCell.swift`. And I was able to test that.

**Comment by pinarol** [comment]
About the [dangermattic warning](https://github.com/wordpress-mobile/WordPress-iOS/pull/23429#issuecomment-2236138202). I ran `swift package resolve` but it didn't change `Package.resolved`. Gravatar was already resolved so i think it's normal.

**Comment by kean** [comment]
> About the https://github.com/wordpress-mobile/WordPress-iOS/pull/23429#issuecomment-2236138202. I ran swift package resolve but it didn't change Package.resolved. Gravatar was already resolved so i think it's normal.

Please disregard this warning as its a false positive. The dependencies are pinned in `./WordPress.xcworkspace/xcshareddata/swiftpm/Package.resolved`. The  `Modules/Packages.resolved` file is in `.gitignore` as it should be. @iangmaia, could you please take a look into this warning?

AppCenter is down. I'll test it once it back up again. Thank you for the PR! 🙇

File: Modules/Sources/WordPressUI/Extensions/Gravatar/Gravatar.swift [hunk 2]
```diff
@@ -1,148 +0,0 @@
...
-    /// - Parameter email: the email associated with the gravatar
-    /// - Returns: hashed email
-    ///
-    /// This really ought to be in a different place, like Gravatar.swift, but there's
-    /// lots of duplication around gravatars -nh
-    private static func gravatarHash(of email: String) -> String {
-        return email
-            .lowercased()
-            .trimmingCharacters(in: .whitespaces)
-            .sha256Hash()
```
Comment (kean) [inline Modules/Sources/WordPressUI/Extensions/Gravatar/Gravatar.swift:109]: (Nit) `sha256Hash` can now also be removed. I left a note to do it later unless you get to it first.

**Comment by iangmaia** [comment]
> > About the [#23429 (comment)](https://github.com/wordpress-mobile/WordPress-iOS/pull/23429#issuecomment-2236138202). I ran swift package resolve but it didn't change Package.resolved. Gravatar was already resolved so i think it's normal.
> 
> Please disregard this warning as its a false positive. The dependencies are pinned in `./WordPress.xcworkspace/xcshareddata/swiftpm/Package.resolved`. The `Modules/Packages.resolved` file is in `.gitignore` as it should be. @iangmaia, could you please take a look into this warning?
> 

Thanks, will look into it! It is indeed a false positive.

File: Modules/Sources/WordPressUI/Extensions/Gravatar/Gravatar.swift (hunk 2 above)
Comment (pinarol) [reply Modules/Sources/WordPressUI/Extensions/Gravatar/Gravatar.swift:109]: Done 👍

**Comment by kean** [comment]
Tested – all looks good :shipit: 
App Center still down .

**Review by kean — APPROVED** [review]

**Pull request merged**